lib/OpenQA/Worker/Job.pm

```diff
@@ -561,27 +562,37 @@ sub _format_reason {
     elsif ($reason eq 'cancel') {
         return undef;    # the result is sufficient here
     }
-    else {
-        return $reason;
-    }
-}
-
-sub _set_job_done {
-    my ($self, $reason, $params, $callback) = @_;
 
-    # Retrieve extended reason if available
-    my $pooldir    = $self->worker->pool_directory;
-    my $state_file = path($pooldir)->child(BASE_STATEFILE);
+    # consider other reasons as os-autoinst specific; retrieve extended reason if available
+    my $state_file = path($self->worker->pool_directory)->child(BASE_STATEFILE);
     try {
         if (-e $state_file) {
-            my $msg = decode_json($state_file->slurp)->{msg};
-            $reason = "$reason: " . $msg unless $msg =~ /\n/;
+            my $state = decode_json($state_file->slurp);
+            die 'top-level element is not a hash' unless ref $state eq 'HASH';
+            if (my $component = $state->{component}) {
+                # prepent the relevant component, e.g. turn "died" into "backend died" or "tests died"
+                $reason = "$component $reason" unless $component =~ qr/$[\w\d]^/;
+            }
+            if (my $msg = $state->{msg}) {
+                # append additional information, e.g. turn "backend died" into "backend died: qemu crashed"
+                $reason = "$reason: $msg" unless $msg =~ /\n/;
+            }
         }
     }
     catch {
-        $reason = "$reason: Corrupted state file could not be read";
+        if ($reason eq 'done') {
+            $reason = "$reason: terminated with corrupted state file";
```

Comment:
Can we find a phrasing that includes a user action? If I would read this as user I would ask "So, what now?"

Reply:
The user might file an os-autoinst bug after checking that everything was setup correctly. Of course the same it true for `Reason: backend died: Migrate to file failed` mentioned as example in the commit message.

Not sure whether it makes sense to include a hint like that. They usually file bugs when they see incompletes and can not make sense of them anyways.

Reply:
I see. But under which circumstances could this fail in this way?

Reply:
From the point of view of openQA this JSON file is provided by an external source (os-autoinst and any user can mess with it) so it makes sense to validate the input here (as it was already the case before my changes) and report possible problems. In this case it would be especially bad to let an exception pass because the worker should in any case mark the job as done.

Reply:
hm, fine